From a9b6db7a117c866c6e9584a550f08a243c97c12d Mon Sep 17 00:00:00 2001
From: Ecco Park <eccopark@broadcom.com>
Date: Wed, 30 Oct 2013 10:31:36 +0900
Subject: [PATCH 256/338] net: wireless: bcmdhd: Fixed a problem of buganizer
 issue (11190960)

	Problem : if PNO scan occurs in Driver after p2p scan,
		  we failed to do PNO scan
	Root cause : In the wl_run_escan function, we still goes into
	p2p scan code even though current scan is PNO which is kind of
	legacy scan.

	Solution : we need to clear the staus of P2P scan before doing
	PNO SCAN.

Signed-off-by: Ecco Park <eccopark@broadcom.com>
(cherry picked from commit ccc9a864869490c267fbb4a5a6c02ff209bbeb2a)
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 drivers/net/wireless/bcmdhd/wl_cfg80211.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/bcmdhd/wl_cfg80211.c b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
index 7bb07b9..3cbc7ea 100644
--- a/drivers/net/wireless/bcmdhd/wl_cfg80211.c
+++ b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
@@ -8080,6 +8080,7 @@ wl_notify_sched_scan_results(struct wl_priv *wl, struct net_device *ndev,
 				wl_clr_drv_status(wl, SCANNING, ndev);
 				goto out_err;
 			}
+			p2p_scan(wl) = false;
 		}
 
 		wl_set_drv_status(wl, SCANNING, ndev);
-- 
1.7.9.5

